rq/worker.py

```diff
@@ -474,8 +474,17 @@ def perform_job(self, job):
         self.set_current_job_id(job.id)
         self.heartbeat((job.timeout or 180) + 60)
 
+        try:
+            func_name = job.func_name
+        except UnpickleError:
+            # Use the public setter here, to immediately update Redis
+            job.set_status(Status.FAILED)
+            self.log.exception('RQ failed to unpickle job %s' % job.id)
```

Comment:
Which leads into my next proposal:  instead of passing `exc_info` to handlers, I'd rather get the `Exception` instance